sycl/include/CL/sycl/queue.hpp

```diff
@@ -22,6 +22,41 @@
 
 #include <utility>
 
+// having _TWO_ mid-param #ifdefs makes the functions very difficult to read.
+// Here we simplify the &CodeLoc declaration to be _CODELOCPARAM(&CodeLoc) and
+// _CODELOCARG(&CodeLoc) Similarly, the KernelFunc param is simplified to be
+// _KERNELFUNCPARAM(KernelFunc) Once the queue kernel functions are defined,
+// these macros are #undef immediately.
+
+// replace _CODELOCPARAM(&CodeLoc) with nothing
+// or :   , const detail::code_location &CodeLoc =
+// detail::code_location::current()
+// replace _CODELOCARG(&CodeLoc) with nothing
+// or :  const detail::code_location &CodeLoc = {}
+
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+#define _CODELOCONLYPARAM(a)                                                   \
```

Comment:
It seems the `_CODELOCONLYPARAM` is defined only if `DISABLE_SYCL_INSTRUMENTATION_METADATA` is not defined while used unconditionally on line 275.

Reply:
Fixed.